### PR TITLE
patch(main): Live Migration, and MSync support

### PR DESCRIPTION
**Pull request by ShivanshVij**
This PR is only for change tracking purposes and should not be merged.

### CODE REVIEW

File: src/vmm/src/arch/x86_64/msr.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ const APIC_BASE_MSR: u32 = 0x800;
 /// Number of APIC MSR indexes
 const APIC_MSR_INDEXES: u32 = 0x400;
 
-/// Custom MSRs fall in the range 0x4b564d00-0x4b564dff
+/// Custom KVM MSRs fall in the range 0x4b564d00-0x4b564dff
```
Comment (pojntfx) [inline src/vmm/src/arch/x86_64/msr.rs:52]: Is this change left over from the PVM MSR changes?

File: src/vmm/src/arch/x86_64/msr.rs [hunk 2]
```diff
@@ -361,6 +361,7 @@ const UNDUMPABLE_MSR_RANGES: [MsrRange; 17] = [
     MSR_RANGE!(HV_X64_MSR_SYNDBG_OPTIONS),
     MSR_RANGE!(HV_X64_MSR_CRASH_P0, 6),
     MSR_RANGE!(HV_X64_MSR_REENLIGHTENMENT_CONTROL, 3),
+    // HV_X64_MSR_TSC_INVARIANT_CONTROL
```
Comment (pojntfx) [inline src/vmm/src/arch/x86_64/msr.rs:364]: See https://github.com/loopholelabs/firecracker/pull/9#discussion_r1583907587

File: tests/integration_tests/functional/test_cpu_template_helper.py [hunk 3]
```diff
@@ -375,3 +375,35 @@ def test_consecutive_fingerprint_consistency(cpu_template_helper, tmp_path):
 
     # Compare them.
     cpu_template_helper.fingerprint_compare(fp1, fp2, None)
+    # Strip common entries.
```
Comment (pojntfx) [inline tests/integration_tests/functional/test_cpu_template_helper.py:378]: Are these changes related to live migration/msync or are they backports from another branch?

File: src/vmm/src/arch/x86_64/msr.rs (hunk 1 above)
Comment (ShivanshVij) [reply src/vmm/src/arch/x86_64/msr.rs:52]: Nope

File: src/vmm/src/arch/x86_64/msr.rs (hunk 2 above)
Comment (ShivanshVij) [reply src/vmm/src/arch/x86_64/msr.rs:364]: Nope

File: tests/integration_tests/functional/test_cpu_template_helper.py (hunk 3 above)
Comment (ShivanshVij) [reply tests/integration_tests/functional/test_cpu_template_helper.py:378]: Backports from main actually

**Comment by pojntfx** [comment]
LGTM! Tested with the drafter snapshotter & runner without any issues (Linux 6.7.0-rc6 & 6.1.89) :)

**Comment by pojntfx** [comment]
Superseded by #11 and #10

**Pull request closed without merging**